ray-operator/main.go

```diff
@@ -140,20 +143,20 @@ func main() {
 	// and that label is provided to the manager cache as a selector for Job resources.
 	selectorsByObject, err := cacheSelectors()
 	exitOnError(err, "unable to create cache selectors")
+	options.Cache.ByObject = selectorsByObject
+
 	if watchNamespaces := strings.Split(watchNamespace, ","); len(watchNamespaces) == 1 { // It is not possible for len(watchNamespaces) == 0 to be true. The length of `strings.Split("", ",")` is still 1.
-		options.Namespace = watchNamespaces[0]
 		if watchNamespaces[0] == "" {
 			setupLog.Info("Flag watchNamespace is not set. Watch custom resources in all namespaces.")
 		} else {
 			setupLog.Info(fmt.Sprintf("Only watch custom resources in the namespace: %s", watchNamespaces[0]))
+			options.Cache.DefaultNamespaces[watchNamespaces[0]] = cache.Config{}
 		}
-		options.NewCache = cache.BuilderWithOptions(cache.Options{SelectorsByObject: selectorsByObject})
 	} else {
-		options.NewCache = func(config *rest.Config, opts cache.Options) (cache.Cache, error) {
```

Comment:
Note: https://sourcegraph.com/github.com/kubernetes-sigs/controller-runtime@2154ffbc22e26ffd8c3b713927f0df2fa40841f2/-/blob/pkg/manager/manager.go?L113-L120

> 	// NewCache is the function that will create the cache to be used
	// by the manager. If not set this will use the default new cache function.
	//
	// When using a custom NewCache, the Cache options will be passed to the
	// NewCache function.
	//
	// NOTE: LOW LEVEL PRIMITIVE!
	// Only use a custom NewCache if you know what you are doing.